Source/ProgressionSystemRuntime/Public/Widgets/PSOverlayWidget.h

```diff
@@ -13,21 +14,45 @@ UCLASS()
 class PROGRESSIONSYSTEMRUNTIME_API UPSOverlayWidget : public UUserWidget
 {
 	GENERATED_BODY()
+
 public:
-	
+	/**
+	* Sets the visibility of the overlay elements and playing fade animation if needed 
+	* @param VisibilitySlate The visibility state (e.g., Visible, Collapsed) to apply to the overlay and icon.
+	* @param bShouldPlayFadeAnimation Defines if the fade animation should be played after or before widget is visible
+	*/
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void SetOverlayVisibility(ESlateVisibility VisibilitySlate, bool bShouldPlayFadeAnimation = false);
+
+protected:
+	/** overrides NativeTick to make the user widget tickable **/
+	virtual void NativeTick(const FGeometry& MyGeometry, float InDeltaTime) override;
+	/** Event to execute when widget is ready */
+	virtual void NativeConstruct() override;
+
+	/**
+	* Play the overlay elements fade-in/fade-out animation.
+	* Uses the internal FadeCurveFloatInternal initialized in NativeConstruct
+	*/
+	UFUNCTION(BlueprintCallable, Category="C++")
+	void TickPlayFadeOverlayAnimation();
```

Comment:
It's under `protected`, so please add `BlueprintProtected` meta, otherwise it's being protected only in code while other blueprints still can call it as `public`

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/20/commits/d8183dbd485677952581c3a062b825b7faca4cb3